pymeasure/display/windows.py

```diff
@@ -414,16 +430,27 @@ def make_procedure(self):
                             " without a InputsWidget type")
         return self.inputs.get_procedure()
 
-    def new_curve(self, results, color=None, **kwargs):
+    def new_curve(self, wdg, results, color=None, **kwargs):
         if color is None:
             color = pg.intColor(self.browser.topLevelItemCount() % 8)
-        return self.plot_widget.new_curve(results, color=color, **kwargs)
+        return wdg.new_curve(results, color=color, **kwargs)
+
+    def new_experiment(self, results, curve=[]):
```

Comment:
I think in principle it is safer to have `None` as a default argument rather than a mutable (like `[]`); if anything is added to this mutable, the default value for next calls of this function will be changed. (In this case I do not see this problem arise, but just as a precaution).

Reply:
Thanks for the head up, I did not realize this potential side effect which is very hard to debug. I will change to None